src/rmq/rmqamqpt/rmqamqpt_frame.cpp

```diff
@@ -66,19 +66,21 @@ Frame::ReturnCode Frame::decode(Frame* frame,
         return Frame::PARTIAL;
     }
 
-    bsl::memcpy(&frame->d_type, &buffer[0], sizeof(frame->d_type));
-
-    bdlb::BigEndianUint16 channel;
-    bsl::memcpy(&channel, &buffer[1], sizeof(channel));
-    frame->d_channel = channel;
-
+    // Decode payload length first to return early if Frame is not full.
     bdlb::BigEndianUint32 payloadLength;
     bsl::memcpy(&payloadLength, &buffer[3], sizeof(payloadLength));
 
-    if ((frameOverhead() + payloadLength) > bufferLen) {
+    if (bufferLen < (frameOverhead() + payloadLength)) {
         return Frame::PARTIAL;
     }
 
+    // Finally decode the skipped parts of the header.
+    bsl::memcpy(&frame->d_type, &buffer[0], sizeof(frame->d_type));
```

Comment:
Side effect: partial `Frame` object will not be modified in any way during decode.